pkg/equality/virtualgateway_types_equality.go

```diff
@@ -0,0 +1,18 @@
+package equality
+
+import (
+	appmeshsdk "github.com/aws/aws-sdk-go/service/appmesh"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+)
+
+func CompareOptionForVirtualGatewayHealthCheckPolicy() cmp.Option {
+	return IgnoreLeftHandUnset(appmeshsdk.VirtualGatewayHealthCheckPolicy{}, "Port")
+}
+
+func CompareOptionForVirtualGatewaySpec() cmp.Option {
```

Comment:
Is the "Port" been assigned by appmesh server-side if it's unset?
If so, you need to use CompareOptionForVirtualGatewaySpec instead of cmpopts.EquateEmpty() in `updateSDKVirtualGateway`